MentorMatchmaker/ApplicantInput.fs

```diff
@@ -0,0 +1,185 @@
+﻿[<RequireQualifiedAccess>]
+module MentorMatchmaker.ApplicantInput
+
+open System
+open type StringSplitOptions
+open FSharp.Data
+
+open Domain
+open MentorMatchmaker
+open Microsoft.FSharp.Core.CompilerServices
+open FsToolkit.ErrorHandling
+
+type MentorshipInformation = CsvProvider<"mentorship_schema_file.csv">
+type InvalidInput = InvalidInput of invalidInput:string * message:string
+
+[<AutoOpen>]
+module Functions =
+
+    let generateWeekList (text:string) startHour endHour = [
+        let weekdays =
+            text.Split([|','; ';'|], TrimEntries ||| RemoveEmptyEntries)
+            |> Array.choose Option.tryParse<DayOfWeek>
+
+        for weekday in weekdays do
+            for hour in startHour .. endHour - 1 do
+                yield { Weekday = weekday; Time = TimeOnly(hour, 0, 0) }
+    ]
+
+    let convertTextToTopic topicText =
+        let deepDiveInFSharpKeywords =
+            [ "Deep";
+              "dive";
+              "investment";
+              "better" ]
+
+        let mobileDevelopmentKeywords =
+            [ "Uno";
+              "Fabulous";
+              "Xamarin";
+              "Mobile";
+              "Mobile development" ]
+
+        let distributedSystemKeywords =
+            [ "Microservices";
+              "Distributed systems";
+              "event sourcing" ]
+
+        let webDevelopmentKeywords =
+            [ "Web";
+              "Elmish";
+              "Fable";
+              "SAFE";
+              "Giraffe";
+              "React";
+              "Feliz";
+              "MVC";
+              "Web development / SAFE stack" ]
+
+        let openSourceKeywords =
+            [ "Contribute to an open source project";
+              "open source" ]
+
+        let matchOnTopicText (topicText: string) =
+            let doesCategoryMatchKeyword (categoryName: string) (keywordList: string list) =
+                keywordList
+                |> List.exists
+                    (fun keyword -> categoryName.Contains(keyword, StringComparison.InvariantCultureIgnoreCase))
+            let category =
+                if topicText[0] = ' ' then
```

Comment:
You fix your form to make that a required field 😉.

TBH it's some of the logic that was there before my changes, might not be necessary since I added `TrimEntries`

Reply:
Can you test it and see whether we can eliminate it from the codebase?

Reply:
Yep, it tested fine. 

cf86f02142cfcf174ca3262c1b7628e50c6591a2